packages/api/src/util/getMetadata.ts

```diff
@@ -0,0 +1,50 @@
+import { getProvider } from '@cennznet/api/util/getProvider';
+import { TypeRegistry } from '@polkadot/types/create';
+import { RpcCore } from '@polkadot/rpc-core';
+import { MetadataVersioned } from '@polkadot/metadata/MetadataVersioned';
+import { Metadata } from '@polkadot/metadata';
+import { ProviderInterface } from '@polkadot/rpc-provider/types';
+import { WsProvider } from '@polkadot/rpc-provider';
+
+const KEEP = ['genericasset', 'cennzx'];
+const MUST = ['system', 'timestamp', 'transactionpayment'];
+export async function getMetadata(
+  provider: string | ProviderInterface,
+  keepModules: string[] = KEEP
+): Promise<Record<string, string>> {
+  const useModules = keepModules.map(module => module.trim().toLowerCase());
+  MUST.map(mandatoryModule => {
+    if (!useModules.includes(mandatoryModule)) {
+      useModules.push(mandatoryModule);
+    }
+  });
+  let providerInterface: ProviderInterface;
+  if (typeof provider === 'string') {
+    providerInterface = getProvider(provider);
+  } else {
+    providerInterface = provider || new WsProvider();
+  }
+  const registry = new TypeRegistry();
+  const rpcCore = new RpcCore('1', registry, providerInterface);
+  const meta: Metadata = await rpcCore.state.getMetadata().toPromise();
+  const magicNumber = meta.magicNumber;
+  const modules = meta.asLatest.modules.toArray();
+
+  const extrinsic = meta.asLatest.extrinsic;
+  const filteredModule = modules.filter((mod) => useModules.find((a) => a === mod.name.toString().toLowerCase()));
+  const filteredModuleMetadataLatest = registry.createType('Vec<ModuleMetadataLatest>', filteredModule);
+  const metadataSlim = registry.createType('MetadataLatest', {
+    extrinsic,
+    modules: filteredModuleMetadataLatest,
+  });
+  const mVersionedSlim = new MetadataVersioned(registry, {
+    magicNumber: magicNumber,
+    metadata: registry.createType('MetadataAll', metadataSlim, 12),
```

Comment:
why `12`? does this need to change? if so when?

Reply:
no.. m going to change this so that we don't need to touch this piece